pkg/services/ngalert/store/alert_rule.go

```diff
@@ -566,22 +535,36 @@ func (st DBstore) GetAlertRulesForScheduling(ctx context.Context, query *ngmodel
 		query.ResultRules = rules
 
 		if query.PopulateFolders {
-			foldersSql := sess.Table("folder").Alias("d").Select("d.org_id, d.uid, d.title, d.parent_uid").
-				Where(`EXISTS (SELECT 1 FROM alert_rule a WHERE d.uid = a.namespace_uid AND d.org_id = a.org_id)`)
-			if len(disabledOrgs) > 0 {
-				foldersSql.NotIn("org_id", disabledOrgs)
-			}
-
-			if err := foldersSql.Find(&folders); err != nil {
-				return fmt.Errorf("failed to fetch a list of folders that contain alert rules: %w", err)
+			query.ResultFoldersTitles = map[ngmodels.FolderKey]string{}
+			uids := map[int64]map[string]struct{}{}
+			for _, r := range rules {
+				om, ok := uids[r.OrgID]
+				if !ok {
+					om = make(map[string]struct{})
+					uids[r.OrgID] = om
+				}
+				om[r.NamespaceUID] = struct{}{}
 			}
-			query.ResultFoldersTitles = make(map[ngmodels.FolderKey]string, len(folders))
-			for _, f := range folders {
-				key := ngmodels.FolderKey{
-					OrgID: f.OrgId,
-					UID:   f.Uid,
+			for orgID, uids := range uids {
+				schedulerUser := accesscontrol.BackgroundUser("grafana_scheduler", orgID, org.RoleAdmin,
```

Comment:
The problem here is that this is supposed to be a multi-org user (it requires per -org permissions) we can extract it to a function but it won't make any difference because we won't be able to pass it around in scheduler.

Reply:
I see. I wonder if we could change UX so user is not guided into the path? Put folder selector next to name selector, above the query or something? @gillesdemey

Reply:
Clicked the wrong comment box...that was meant for the one about testing API 
https://github.com/grafana/grafana/pull/81214#discussion_r1475313087